lib/helpers/broadcast.js

```diff
@@ -80,9 +123,46 @@ function parseBroadcastFromContentfulEntry(contentfulEntry) {
   return data;
```

Comment:
We would also have to return with a Promise from this function so that both `legacy` and `new` broadcasts will return Promises from `parseBroadcastFromContentfulEntry`.